src/npm.py

```diff
@@ -0,0 +1,70 @@
+from glob import glob
+import os
+import re
+import sys
+import json
+import frontmatter
+import urllib.request
+
+DEFAULT_TAG_TEMPLATE = (
+    "{{major}}{% if minor %}.{{minor}}{% if patch %}.{{patch}}{%endif%}{%endif%}"
+)
+REGEX = r"^(?:(\d+\.(?:\d+\.)*\d+))$"
+
+
+def fetch_releases(npm_id, regex):
+    releases = {}
+
+    if not isinstance(regex, list):
+        regex = [regex]
+
+    url = f"https://registry.npmjs.org/{npm_id}"
+    with urllib.request.urlopen(url, data=None, timeout=5) as response:
+        data = json.loads(response.read().decode("utf-8"))
+        for version in data["time"]:
+            R = data["time"][version]
+            matches = False
+            for r in regex:
+                if re.match(r, version):
+                    matches = True
+            if matches and R:
+                d = ""
+                for c in R:
```

Comment:
R is an ISO 8601 timestamp. It would be simpler to split with `T` and keeping the first part.

Reply:
Done.